### PR TITLE
fixed colors in xy for common usage cases + unittest

### PR DESCRIPTION
**Pull request by albalu**
This should address Issue #188 . One may check the results with the following simple but relatively common examples.

from matminer import PlotlyFig
import pandas as pd
data = {'a': [1, 2, 3, 4],
        'b': [2.5, -1, 2, 6],
        'c': [2, 4, 6, 8]}
df = pd.DataFrame.from_dict(data)
pf = PlotlyFig(df)
pf.xy([('a', 'b'), ('a', 'c')], modes='lines+markers', colors=['green', 'blue'])
pf.xy([('a', 'b'), ('a', 'c')], modes='lines+markers', colors='red')
pf.xy([('a', 'b'), ('a', 'c')], modes='lines+markers', colors='c', colorscale='Rainbow')
pf.xy([('a', 'b'), ('a', 'c'), ('a', 'a')], modes='markers', colors='a')
pf.xy([('a', 'b'), ('a', 'c'), ('a', 'a')], modes=['markers', 'lines', 'markers+lines'],
      colors=[data['c'], 'green', 'c'])
pf.xy([('a', 'b'), ('a', 'c')], modes=['lines', 'markers'], colors=['blue', 'b'])

### CODE REVIEW

**Comment by ardunn** [comment]
@albalu Given that changing even very small parts of FigRecipes can cause problems in lots of plots, could you just confirm that all the matminer_examples xy plots are still correctly created using these changes? Once confirmed I'll merge

**Comment by albalu** [comment]
ok I just confirmed that xy examples work as expected

**Comment by albalu** [comment]
@ardunn  I think it's the default line width of the markers. Line width default is 2. BTW I don't see what you changed.

File: matminer/figrecipes/plot.py [hunk 1]
```diff
@@ -480,14 +477,45 @@ def xy(self, xy_pairs, colors=None, color_range=None, labels=None,
...
+            labels = self._data_from_str(labels)
             labels = [labels] * len(data)
         else:
-            labels = [self.data_from_col(l) for l in labels]
-        markers = markers or [{'symbol': 'circle', 'line': {'width': 1,
-                                                            'color': 'black'}}
-                              for _ in data]
+            labels = [self._data_from_str(l) for l in labels]
+        markers = markers or [{'symbol': 'circle',
+                               'line': {'width': 1,'color': 'black'}
```
Comment (ardunn) [inline matminer/figrecipes/plot.py]: 'width': 1 is what I am talking about

Comment (ardunn) [reply matminer/figrecipes/plot.py]: @albalu

**Comment by ardunn** [comment]
@albalu nvm, it was a marker outline not the linewidth

**Pull request merged**